### PR TITLE
[WFLY-16796] LocalEjbReceiver remove ContextData that was removed in …

### PR DESCRIPTION
**Pull request by bmaxwell**
…EJB and test case for WFLY-16796 / WFLY-16567

https://issues.redhat.com/browse/WFLY-16796

### CODE REVIEW

**Comment by bmaxwell** [comment]
This includes tests for it and previous issue, WFLY-16567 is already merged.  It needs component upgrades for WF Http Client, looks like WFLY-16741 is merged.  And ejb client is merged WFLY-16740.  So everything should be merged so that the included tests pass.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/contextdata/TestRemote.java [hunk 1]
```diff
@@ -26,6 +26,6 @@
 @Remote
 public interface TestRemote {
 
-  String invoke();
+    UseCaseValidator invoke(UseCaseValidator useCaseValidator) throws TestException;
```
Comment (chengfang) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/contextdata/TestRemote.java:29]: `invoke()` is used by other test class and should be kept.

https://github.com/wildfly/wildfly/blob/3b0cdbe49beb445f8b93f972b4a6b8a4887d1f15/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/contextdata/ClientInterceptorReturnDataEjbOverHttpTestCase.java#L110

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/contextdata/TestSLSB.java [hunk 2]
```diff
@@ -30,15 +30,14 @@
...
 
-    private static final String MESSAGE = "bean context data";
-
     @Resource
     SessionContext sessionContext;
 
+    @Override
+    public UseCaseValidator invoke(UseCaseValidator useCaseValidator) throws TestException {
 
-    public String invoke() {
```
Comment (chengfang) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/contextdata/TestSLSB.java:39]: same as above.

Comment (bmaxwell) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/contextdata/TestSLSB.java:39]: @chengfang thanks !

**Comment by bstansberry** [comment]
@chengfang Is this ok now? If so, should it wait for WF 28?

**Comment by chengfang** [comment]
this new test `org.jboss.as.test.integration.ejb.remote.contextdata.ClientInterceptorReturnDataEjbOverHttpTestCase.testInvokeWithClientInterceptorData` failed due to NPE consistently. I'll take a look.

**Comment by bstansberry** [comment]
oh, sorry; i didn't scroll down enough to see the test results!

**Comment by bmaxwell** [comment]
@chengfang thanks, I looked but it did not make sense how it could return null on the ejb lookup and the test had passed on my machine previously, so not sure what changed.  I will look again next Monday when I'm back home and have my work machine.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/contextdata/ClientInterceptor.java [hunk 3]
```diff
@@ -42,20 +41,25 @@ public ClientInterceptor() {
...
   @Override
   public void handleInvocation(EJBClientInvocationContext context) throws Exception {
-    HashSet<String> keys = new HashSet<>();
-    keys.add("data1");
-    keys.add("data2");
-    context.getContextData().put("jboss.returned.keys", keys); //TODO: replace with call to org.jboss.ejb.client.EJBClientInvocationContext#addReturnedContextDataKey
-    context.getContextData().put("clientData", DATA);
+
+      logger.debugf("%s: ContextData: %s", "handleInvocation", context.getContextData());
+      ((UseCaseValidator)context.getParameters()[0]).test(UseCaseValidator.InvocationPhase.CLIENT_INT_HANDLE_INVOCATION, context.getContextData());
```
Comment (chengfang) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/contextdata/ClientInterceptor.java:46]: the above `context.getParameters()` call can return null, and when accessing its 0th element, NPE is triggered.

The test ejb remote interface `TestRemote` contains 2 methods, one has param `UseCaseValidator` and the other does not. When the no-param method `invoke()` is called, the above NPE condition occurred:

`org.jboss.as.test.integration.ejb.remote.contextdata.ClientInterceptorReturnDataEjbOverHttpTestCase#testInvokeWithClientInterceptorData`, around line 110.

Comment (chengfang) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/remote/contextdata/ClientInterceptor.java:46]: @bmaxwell did you get a chance to look at it?

**Comment by TomasHofman** [comment]
@bmaxwell the test where the `context.getParameters()` returns null is a different test class than those you modified. You likely just overlooked it. (This remaining test class still calls the original no param method.)

I created a copy of this PR and added another commit to fix the remaining test class here: https://github.com/wildfly/wildfly/pull/16431

Can you review it?

**Comment by TomasHofman** [comment]
@bmaxwell can you fix the remaining test here, or review https://github.com/wildfly/wildfly/pull/16431 where I tried to fix it?

**Comment by bmaxwell** [comment]
@TomasHofman thanks, I've updated my PR, you can use it or yours which ever.  I reviewed your changes where the one test was returning null and then the clean up and null checks added, looks good to me.

**Comment by TomasHofman** [comment]
@chengfang or @bstansberry can you re-review?

**Comment by bstansberry** [comment]
I'll defer to the EJB team on this one.

**Comment by TomasHofman** [comment]
Thanks! I closed my PR, we will use this one.

**Review by chengfang — APPROVED** [review]

**Comment by TomasHofman** [comment]
For the record, downstream PR has been merged.

**Comment by bstansberry** [comment]
Thanks @bmaxwell and @chengfang

**Pull request merged**